### ISSUE TITLE
Alignment Offset (Portal={true})

### ISSUE THREAD

**Issue by danglez83** [szhsin/react-menu#1009]
React/React-dom version: v18.2.0
React-Menu version: v4.0.0

## Describe the bug
When setting Portal={true} and align='end' (could occur with the other values as well), the **first time** a menu is dropped down, there is a slight offset to the position of the menu; the position is accurate in subsequent attempts to drop down the menu. Happens with every menu in my app if those settings are set.

## To Reproduce

1. Created a basic menu
2. Set align = 'end'
3. Set portal = {true}
4. Dropdown the menu once (offset), dropdown again (fixed)

## Code snippets/CodeSandbox examples/Screenshots
[https://ibb.co/nDyp07X](url)

![offset](https://github.com/szhsin/react-menu/assets/42001195/c35dbbc4-8fbc-487f-855b-5c35970ab72e)

**Comment by szhsin** [comment]
Hello @danglez83 , your screen recording did demonstrate menu failed to align properly on the first drop down. However, I was not able to reproduce it on my end, neither in a [sandbox example](https://codesandbox.io/s/react-menu-starter-forked-hll62r?file=/src/App.js) nor a nextJS project. 

I'd appreciate it if you could provide a sandbox environment that can showcase the issue and help us investigate it.

**Comment by danglez83** [comment]
Here you go @szhsin, hopefully this helps. I tried to pare my exterior code down as best I could. The issue appears to be when the height is set to 100% on a parent.

[https://codesandbox.io/s/react-menu-starter-forked-lm2wp7?file=/src/App.js](url)

**Comment by danglez83** [comment]
Sorry @szhsin, accidentally closed the issue. Hopefully my sandbox helps!

**Comment by szhsin** [comment]
Thanks @danglez83, your sandbox really helps! Will be looking into the issue.

**Comment by sohamsshah** [comment]
Hi @danglez83,
I solved this issue by adding `unmountOnClose` prop to the `Menu` component. Let me know if it works for you. Zheng, I would like to understand why this worked and is it a hacky way? 

Side note: Thanks for building this awesome lib <3 All Hail Open Source

**Comment by danglez83** [comment]
> Hi @danglez83, I solved this issue by adding `unmountOnClose` prop to the `Menu` component. Let me know if it works for you. Zheng, I would like to understand why this worked and is it a hacky way?
> 
> Side note: Thanks for building this awesome lib <3 All Hail Open Source

Just an FYI @sohamsshah, adding that prop to my app did not _fix_ the issue, which kind of makes sense since this issue only happens on initial menu dropdown.

**Comment by szhsin** [comment]
Hey folks, it turns out the issue was caused by a vertical scrollbar.

When a menu is to be positioned, we use two render passes. In the first pass menu is placed at a default position which in some cases might cause browser to display a scrollbar, for example, page has 100% height. Menu position is then calculated and repositioned in the second pass, taking into account the scrollbar width (15px on macOS). Now menu is positioned close to anchor and scrollbar is no longer needed for the page. This results in a 15px horizontal misalignment. All these steps are finished before browser repaints the screen so that we don't see the scrollbar in the end.

On the second dropdown, menu is already positioned close to anchor, thus browser doesn't create scrollbars as in the first dropdown.

I fixed the bug by giving menu a large negative position by default, which ensures a scrollbar is not created by the browser.

Again, thanks @danglez83 for bringing this issue to our attention.

To @sohamsshah 
> I solved this issue by adding `unmountOnClose` prop to the Menu component. Let me know if it works for you. Zheng, I would like to understand why this worked and is it a hacky way?

It didn't solve the issue but make the misalignment appear on every dropdown instead of only in the first one. When `unmountOnClose` prop is set, menu position is reset to the default position every time before being opened.
